src/grpc/enrollment.rs

```diff
@@ -447,36 +447,118 @@ impl EnrollmentServer {
             request.pubkey, user.username, user.id
         );
 
-        let device = Device::new(
-            request.name.clone(),
-            request.pubkey.clone(),
-            enrollment.user_id,
-        );
-        debug!(
-            "Creating new device for user {}({:?}) {device:?}.",
-            user.username, user.id,
-        );
-
         let mut transaction = self.pool.begin().await.map_err(|err| {
             error!("Failed to begin transaction: {err}");
             Status::internal("unexpected error")
         })?;
-        let device = device.save(&mut *transaction).await.map_err(|err| {
-            error!(
-                "Failed to save device {}, pubkey {} for user {}({:?}): {err}",
-                request.name, request.pubkey, user.username, user.id,
+
+        let (device, network_info, configs) = if let Some(device_id) = enrollment_token.device_id {
+            debug!(
+                "A device with ID {device_id} is attached to a received enrollment token, trying to finish it's configuration instead of creating a new one."
```

Comment:
```suggestion
                "A device with ID {device_id} is attached to a received enrollment token, trying to finish its configuration instead of creating a new one."
```